File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int i = 0, j = 0, curr0Idx = -1, maxLen = 0;
        while (j < nums.size()) {
            if (nums[j] == 0) {
                i = curr0Idx + 1;
                curr0Idx = j;
            }
            j++;
            maxLen = max(maxLen, j - i);
        }
        return maxLen - 1;
    }
};

File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        nums = [0] + nums + [0]
        n = len(nums)
        dp = []
        maxi = 0
        tmp = 0
        for i in range(n):
            if nums[i] == 0:
                dp.append(tmp)
                tmp = 0
            else:
                tmp += 1
        
        for i in range(1, len(dp)):
            maxi = max(maxi, dp[i] + dp[i - 1])

        return maxi - 1 if maxi == n - 2 else maxi